misk/src/main/kotlin/misk/web/interceptors/RealLogRateLimiter.kt

```diff
@@ -0,0 +1,43 @@
+package misk.web.interceptors
+
+import misk.ratelimit.RateLimiter
+import misk.web.interceptors.LogRateLimiter.LogBucketId
+import java.util.concurrent.ConcurrentHashMap
+import java.util.concurrent.TimeUnit
+import javax.inject.Inject
+import javax.inject.Singleton
+
+/**
+ * Holds rate limiters for logging success and error responses. There is a rate limiter for every
+ * action and service that calls it, for both success and error. The rate limiters are created
+ * according to the value set in [LogRequestResponse] annotation.
+ */
+@Singleton
+class RealLogRateLimiter @Inject constructor(
+  private val rateLimiterFactory: RateLimiter.Factory
+): LogRateLimiter {
+  private val actionToRateLimits = HashMap<String, RateLimitValues>()
+  private val rateLimiters = ConcurrentHashMap<LogBucketId, RateLimiter>()
+
+  override fun addRateLimits(actionClass: String, rateLimiting: Long, errorRateLimiting: Long) {
+    actionToRateLimits.put(actionClass, RateLimitValues(rateLimiting, errorRateLimiting))
+  }
+
+  override fun tryAcquire(bucketId: LogBucketId): Boolean {
+    val rateLimitVals = actionToRateLimits.get(bucketId.actionClass)!!
+    val bucketQps =
+      if (bucketId.isError) rateLimitVals.errorRateLimiting else rateLimitVals.rateLimiting
+    if (bucketQps == 0L) {
+      // If bucketQps is zero, it was specified by the service owner in the [LogRequestResponse]
+      // annotation meaning rate limiting on logs should be off
+      return true
+    }
+    val rateLimiter = rateLimiters.getOrPut(bucketId) { rateLimiterFactory.create(bucketQps) }
+    return rateLimiter.tryAcquire(1, 0L, TimeUnit.MILLISECONDS)
```

Comment:
This doesn't look right, shouldn't the window be 1 second?

Reply:
Ah yeah, fixing